gdnative-core/src/core_types/rid.rs

```diff
@@ -84,3 +84,33 @@ impl PartialOrd for Rid {
         }
     }
 }
+
+#[cfg(feature = "serde")]
+mod serde {
+    use super::*;
+    use ::serde::{de, ser, Deserialize, Deserializer, Serialize, Serializer};
+
+    impl Serialize for Rid {
+        #[inline]
+        fn serialize<S>(&self, _serializer: S) -> Result<<S as Serializer>::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            Err(ser::Error::custom(
```

Comment:
I wonder if it's better to serialize them as null RIDs (`Rid::default()`), rather than preventing the serialization of types containing them (at any depth) wholesale? Since it isn't possible to specify fields to skip when using the serde impls. Not sure if this will actually have any impact, though.

Reply:
Yeah, I'm not happy about the idea of totally *preventing* [de]serialization of any type containing an `RID`... just not sure how best to handle it otherwise. Should it really just deserialize into a null RID, or somehow try to figure out the type of the property and select the correct server to create a new one from? Or should it recursively serialize the type that the RID refers to? (this seems like it could be quite handy, but also very difficult to implement and avoid infinite loops, and some RID's will refer to non-`Resource` types like `CanvasItem`)

Reply:
> select the correct server to create a new one from

Having a valid but empty resource is less useful than having a nil: in the latter case at least you *know* you have a nil.

> recursively serialize the type that the RID refers to

You're right that this is much more complicated and faces the same problem as the object impls. Resources can also be large and suboptimal in textual formats, like textures and meshes. This is most likely more trouble than it's worth.